public/pages/utils/anomalyResultUtils.ts

```diff
@@ -858,3 +937,245 @@ export const filterWithHeatmapFilter = (
   }
   return filterWithDateRange(data, heatmapCell.dateRange, timeField);
 };
+
+export const getTopAnomalousEntitiesQuery = (
+  startTime: number,
+  endTime: number,
+  detectorId: string,
+  size: number,
+  sortType: AnomalyHeatmapSortType
+) => {
+  return {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              [AD_DOC_FIELDS.ANOMALY_GRADE]: {
+                gt: 0,
+              },
+            },
+          },
+          {
+            range: {
+              data_end_time: {
+                gte: startTime,
+                lte: endTime,
+              },
+            },
+          },
+          {
+            term: {
+              detector_id: detectorId,
+            },
+          },
+        ],
+      },
+    },
+    aggs: {
+      [TOP_ENTITIES_FIELD]: {
+        nested: {
+          path: ENTITY_FIELD,
+        },
+        aggs: {
+          [TOP_ENTITY_AGGS]: {
+            terms: {
+              field: ENTITY_VALUE_PATH_FIELD,
+              size: size,
+              ...(sortType === AnomalyHeatmapSortType.SEVERITY
```

Comment:
What if user use `AnomalyHeatmapSortType.OCCURRENCES`  ?

Reply:
The default order is sorting by `doc_count` descending, aka by OCCURRENCES, as per [Elastic doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html#search-aggregations-bucket-terms-aggregation-order). Thus, there is no need to specify query for `OCCURRENCES` case